tests/virt_autotest/virtual_network_utils.pm

```diff
@@ -0,0 +1,136 @@
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_setup, 900);
+    }
+}
+
+sub restore_standalone {
+    #File standalone was installed from qa_test_virtualization package
+    my $standalone_path = "/usr/share/qa/qa_test_virtualization/shared/standalone";
+    assert_script_run("source $standalone_path", 60) if (script_run("[[ -f $standalone_path ]]") == 0);
+}
+
+sub destroy_standalone {
+    #File cleanup was installed from qa_test_virtualization package
+    my $cleanup_path = "/usr/share/qa/qa_test_virtualization/cleanup";
+    assert_script_run("source $cleanup_path", 60) if (script_run("[[ -f $cleanup_path ]]") == 0);
+}
+
+sub restart_libvirtd {
+    is_sle('>11') ? systemctl 'restart libvirtd' : script_run("service libvirtd restart");
+}
+
+sub restore_guests {
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, 30, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define /tmp/$_.xml");
+        script_run("virsh start $_");
+        script_run("rm -rf /tmp/$_.xml");
+    }
+}
+
+sub restore_network {
+    my $virt_host_bridge = shift;
+    my $network_mark     = "/etc/sysconfig/network/ifcfg-$virt_host_bridge.new";
+    if (script_run("[[ -f $network_mark ]]") == 0) {
+        assert_script_run("ip link set dev $virt_host_bridge down", 60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_destroy, 900);
```

Comment:
Same here, just do it natively - this decreases the readability.

Reply:
There is the background， there was not created host bridge network interface (br0) as default on worker (OSD), so, after finished libvirt virtual network test, restore the testing environment as restore_network step under under virtual_network_utils. pm.

Reply:
Please remove those two lines:
```
my $bridge_destroy = <<'EOF';
#!/bin/bash
```

Reply:
Had updated code changed